### DISCUSSION TITLE
Story for mobile auth + OAuth

### DISCUSSION THREAD

**Discussion by terabyte128** [fastapi-users/fastapi-users#1451] [Ideas]
Hello! I'm wondering if fastapi-users has any story for how to support native apps with OAuth? 

From what I can tell, these scenarios are already supported:
* Native app auth, with local users: The native app can call the login endpoint, get a Bearer token, and include it on subsequent request.
* Web app, with OAuth: The web app redirects the user to the provider. The user logs in with the provider, the provider makes a callback, fastapi-users exchanges a code for an access token, and eventually returns a token.

But with a native app, we've left the app to go to the browser, so the app doesn't have a way to access the token from the response. We need something like a callback from our web backend. I guess this would imply making fastapi-users into something like an OAuth _server_ as well as a client. Any thoughts on how this might work, or if there are plans to implement it?

**Comment by vallard** [comment]
ah, I just had this same question. It sounds like this is not implemented then correct?
